File: my-first-app/app/src/main/java/org/feup/ddmm/myfirstapp/FirstActivity.java
package org.feup.ddmm.myfirstapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class FirstActivity extends AppCompatActivity {

    public static final String EXTRA_MESSAGE = "ddmm.MESSAGE";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_first);
    }

    public void btSendOnClick(View view) {
        Intent intent = new Intent(this, SecondActivity.class);
        EditText editText = findViewById(R.id.edit_message);
        String message = editText.getText().toString();
        intent.putExtra(EXTRA_MESSAGE, message);
        startActivity(intent);
    }
}

File: MyFirstApp/app/src/main/java/org/feup/ddmm/myfirstapp/FirstActivity.kt
package org.feup.ddmm.myfirstapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class FirstActivity : AppCompatActivity() {

    companion object {
        val EXTRA_MESSAGE: String = "ddmm.MESSAGE"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_first)

        findViewById<Button>(R.id.button_send).setOnClickListener {
            var intent = Intent(this, SecondActivity::class.java)
            val editText = findViewById<EditText>(R.id.edit_message).text.toString()
            intent.putExtra(EXTRA_MESSAGE, editText)
            startActivity(intent)
        }
    }
}
